app/vmalert/alerting.go

```diff
@@ -391,12 +393,22 @@ func (ar *AlertingRule) Exec(ctx context.Context, ts time.Time, limit int) ([]pr
 				ar.logDebugf(ts, a, "PENDING => DELETED: is absent in current evaluation round")
 				continue
 			}
+			// check if alert should keep StateFiring if rule has `keep_firing_for` field
```

Comment:
Should we reflect this behavior in` ExecRange`, so rules replay would respect the new setting ?

Reply:
I'm not sure, I assume `keep_firing_for` is used to make alert stabilized, prevent vmalert from sending amounts of resolved messages to alertmanager then to users. 
But replay is used to backfill data, produce ALERT and ALERT_FOR_STATE, if we apply `keep_firing_for` on that, it might be difficult to understand why ALERT is firing with present `alertstate` label when compares to the results of querying pure expression of alert.
Maybe we can note that replay doesn't support this field?

Reply:
replay mode should produce identical or almost identical (because it uses /query_range) results to the real results. replay can be used to [validate whether the given alert config](https://victoriametrics.com/blog/rules-replay/)(see retroactive alerts section) has triggered or how frequently it was in some state in the past. So I think we should support that unless there is a fundamental obstacle preventing us from doing this

Reply:
Alright, will do)